src/string.cr

```diff
@@ -903,23 +905,23 @@ class String
     end
   end
 
-  def byte_slice(start : Int)
+  def byte_slice(start : Int) : String
     byte_slice start, bytesize - start
   end
 
-  def codepoint_at(index)
+  def codepoint_at(index) : Int
     char_at(index).ord
   end
 
-  def byte_at(index)
+  def byte_at(index) : Int
     byte_at(index) { raise IndexError.new }
   end
 
-  def byte_at?(index)
+  def byte_at?(index) : (UInt8 | Int8 | UInt16 | Int16 | Int32 | Int64 | UInt32 | UInt64)?
```

Comment:
I'm quite skeptical about these annotations, but why does this return such a strange type?